onadata/libs/utils/csv_import.py

```diff
@@ -251,9 +251,43 @@ def submit_csv(username, xform, csv_file, overwrite=False):
     xl_datetime_columns = [
         dt.get('name') for dt in x_json.get('children')
         if dt.get('type') in XLS_DATETIME_FIELDS]
+    xl_integer_columns = [
+        dt.get('name') for dt in x_json.get('children')
+        if dt.get('type') == 'integer'
+    ]
+    xl_decimal_columns = [
+        dt.get('name') for dt in x_json.get('children')
+        if dt.get('type') == 'decimal'
+    ]
 
     try:
         for row in csv_reader:
+            for key in xl_integer_columns:
+                # For some reason Integer fields are not validated
+                # We check that the passed in data is a valid interger
+                val = row.get(key, '')
+                if val:
+                    try:
+                        interger = int(val)
+                    except ValueError:
+                        raise Exception(
+                                'Unknown interger format: {}'.format(val))
+                    else:
+                        row.update({key: interger})
+
+            for key in xl_decimal_columns:
+                # For some reason Decimal fields are not validated
+                # We check that the passed in data is a valid decimal
+                val = row.get(key, '')
+                if val:
+                    try:
+                        decimal = float(row.get('key', ''))
+                    except ValueError:
+                        raise Exception(
```

Comment:
why are we raising an exception here, whereas below we are passing on the `ValueError`?

Reply:
My assumption on the reason this we were passing it before on `date` and `datetime` columns is since we assumed if a `ValueError` was raised the string format was of the correct format that being `yyyy-MM-dd` or the respective `datetime` format.

Reply:
We'll no longer be passing on the `date` and `datetime` checks below too. As they allow strings like `asdasd` to be passed in `datetime` and `date` columns.

Reply:
Cool, I think we should abstract a function for all of these conditionals then, so we're forced to maintain the same behavior